### PR TITLE
Create Application to Create Index and Search

### PR DESCRIPTION
**Pull request by fuzzybrows**
#### What does this PR do?
* Accepts a JSON file Upload
* Checks for file validity
* Creates an Index of Words from Uploaded File
* Provides Search functionality for Created Index 
#### Description of Task to be completed?
This Program Accepts a JSON file from the User and creates an Index of words based contained in the file provided the contents of the file meet the defined format.
The User is then able to Search for one or more words within the created index. And the application displays the locations of the queried words if they exist
#### How should this be manually tested?
By uploading JSON files and Clicking the buttons
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
N/A
#### Screenshots (if appropriate)
screenshot from 2016-12-05 23-04-27
#### Questions:
N/A

### CODE REVIEW

File: font-awesome/scss/font-awesome.scss [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/*!
```
[the remark this answers is not in the document]
Comment (oredavids) [reply font-awesome/scss/font-awesome.scss:1]: I would suggest using a CDN for your Font-Awesome Library particularly if you are not editing the default files, of which, I counted at least 38. So you would have a neater and more efficient code.

File: jasmine/spec/inverted-index-test.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+// tests file
+describe('Read Book Data', () => {
+  it('Should return \'true\' for Valid JSON file', () => {
+    expect(newInstance.validateFile(JSON.stringify(validFile))[0])
+    .toEqual(true);
+  });
+  const report = 'Empty File Please Upload a Valid Json File';
+  it(`Should return ${report} for an upload with no data`, () => {
+    expect(newInstance.validateFile(emptyFile)[1])
+    .toEqual('Empty File Please Upload a Valid Json File');
```
Comment (oredavids) [inline jasmine/spec/inverted-index-test.js]: Nitpick: `.toEqual(report)` should suffice here since you already have the string stored in the variable `report`.

**Comment by danielonodje** [comment]
src/inverted-index.js
Line 16: this.vbook <-- Variable name is vague
Line 29 title <-- varible is actually an a numerical array index
Lines 30 - 43 -> Make titleIndex[title] a variable
Lines 30 - 43 -> can be made shorter and clearer  with a ternary e.g
```
    arr[index] =  arr[index] === undefined ? {} : array[index];
    arr[index].title = title;
```
Lines 67 - 69 Can be chained e.g
```
    terms.toLowerCase
       .match(/regex/)
       .forEach(function)
```
Line 69 sItem <-- non descriptive variable name

validateFile function
     multiple checks for empty file.
    
Good code formatting and docBlock structure

File: jasmine/spec/inverted-index-test.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  });
+  it('Should confirm that the string key is matched to correct object', () => {
+    expect(newInstance.createIndex(validFile).and[0]).toEqual(true);
+  });
+  it('Should confirm correct index is returned', () => {
+    expect(newInstance.getIndex().and[0]).toEqual(true);
+  });
+});
+
+describe('Test Search', () => {
```
Comment (oredavids) [inline jasmine/spec/inverted-index-test.js]: You are required to: `16. Write a new test suite called "Search index".` Hence, you might want to rename this Test Suite

**Comment by oredavids** [comment]
- Write more tests to check for edge cases
- Make Variable names descriptive
